ERCS/erc-7914.md

```diff
@@ -0,0 +1,157 @@
+
+## Motivation
+
+Because Native transfers must be initiated by the calling account, spenders cannot pull Native tokens without executing code on a user's wallet. This presents a problem in intent-based systems, where transactions are expected to be sent on behalf of users rather than initiated by the users themselves.
+
+With smart contract wallets and ERC-7702 delegated EOAs, wallets that support code execution enable a way for spenders to request Native transfers. This EIP defines an interface for both the approval and transfer mechanisms on the account, making authorization-based native transfers possible without requiring the user to initiate the transaction.
+
+## Specification
+
+The contract code MUST implement the following function:
+
+```solidity
+function transferFromNative(address from, address recipient, uint256 amount) public returns (bool);
+```
+
+- `from` : The account that the ETH transfer should originate from. If this function is implemented on the spender’s contract account, then it should check `from` is set to `address(this)`. If `from` is not `address(this)` the contract COULD forward the call. Note that specifically parameterizing a `from` address allows for flexibility, and other types of integrations with this standard.
+- `recipient`: The recipient of the Native transfer.
+- `amount`: The amount of Native to transfer.
+
+SHOULD early return when amount == 0, otherwise anyone can execute any fallback function on any contract on behalf of the user without explicit approval.
+
+MUST verify whether `msg.sender` (the caller) is authorized to execute the Native transfer.
+
+MUST handle the Native transfer and return false on failure.
+
+MUST emit the following event on success:
+
+```solidity
+event TransferFromNative(address indexed from, address indexed to, uint256 value);
+```
+
+### Persistent Approval
+
+An approval system similar to the ERC-20 `approve` mechanism MUST be implemented to allow users to specify trusted accounts that can perform Native transfers on their behalf.
+
+To modify this approval, the following function MUST be implemented.
+
+```solidity
+function approveNative(address spender, uint256 amount) external returns (bool);
+```
+
+- `spender`: The address being authorized.
+- `amount`: The maximum Native amount the `spender` can transfer.
+
+This function SHOULD only be callable by the owner of the account.
+
+The following event MUST be emitted when approvals are updated:
+
+```solidity
+event ApproveNative(address indexed owner, address indexed spender, uint256 value);
+```
+
+### Optional Transient Approval
+
+For cases where approval should only persist within a single transaction, a transient approval mechanism MAY be used. This allows a user to grant temporary approval to a `spender` for a single transaction, without modifying persistent storage.
```

Comment:
I think need clarity on which native allowance to deduct first. Would prefer transient to be deducted first.

Reply:
Ah that is interesting. Yeah definitely needs some outline for order of operations. My gut impl would be to default first to transient like you said. 

I'm not sure how we should handle the case then if there is more requested than the transient amount allowed. Do you then pull from the persisted allotment? It feels like that may be user error of the transient allotment if the caller didn't request enough to be allowed transiently?

Reply:
i think it would be cleaner if we just separated them - `transferFromNative` and `transferFromNativeTransient`. that way it's easier to know which allowance to subtract from